py3status/modules/mpd_status.py

```diff
@@ -110,12 +110,30 @@ class Py3status:
     state_play = '[play]'
     state_stop = '[stop]'
 
+    def _connection(self):
+        try:
+            if self.connection is None:
+                self.connection = MPDClient()
+                self.connection.connect(host=self.host, port=self.port)
+                if self.password:
+                    self.connection.password(self.password)
+            return self.connection
+        except (socket.error, ConnectionError, CommandError) as e:
+            self.connection = None
+            raise e
+
+    def _mpd_disconnect(self):
```

Comment:
Consolidate `_mpd_disconnect` into `_get_mpd`, eg...
```
    def _get_mpd(self, disconnect=False)
        if not disconnect:
            try:
                ... etc... etc...
            ... etc... etc...
        else:
            self.client.disconnect()
            self.client = None
```
```
Use: self._get_mpd() ; self._get_mpd(disconnect=True)
```

Reply:
again, will do. but why do you consider that better?

Reply:
(Opinion) From what I gather, it's all client/socket related... We can avoid that lone method where its purpose is no more than to disconnect the client. I put them together because I think they would fare better together... and we can use same name everywhere... Also, to help making `mpd_status` more readable.